client/src/axiom/axiom.ts

```diff
@@ -38,33 +43,47 @@ export class Axiom<T> {
     });
   }
 
-  async setParams(params: AxiomV2ClientParams) {
-    this.params = { 
-      ...this.params, 
-      ...params, 
+  setOptions(options: AxiomV2ClientOptions) {
+    this.options = { 
+      ...this.options,
+      ...options,
     };
   }
 
+  setCallback(callback: {
+    target?: string;
+    extraData?: string;
+  }) {
+    this.callback = {
+      ...this.callback,
+      ...callback,
+    }
+  }
+
   async prove(input: RawInput<T>): Promise<AxiomV2SendQueryArgs> {
     await this.axiomCircuit.run(input);
 
     const caller = this.config.privateKey !== undefined ? 
       privateKeyToAccount(this.config.privateKey as `0x${string}`).address as string : 
-      this.params?.caller ?? "";
+      this.options?.caller ?? "";
 
-    const defaultOptions: AxiomV2QueryOptions = {
-      refundee: this.params?.options?.refundee ?? caller,
+    const options: AxiomV2ClientOptions = {
+      ...this.options,
+      refundee: this.options?.refundee ?? caller,
     }
 
     return await this.axiomCircuit.getSendQueryArgs({
-      callbackTarget: this.params?.callback?.target ?? zeroAddress,
-      callbackExtraData: this.params?.callback?.extraData ?? zeroHash,
+      callbackTarget: this.callback.target,
+      callbackExtraData: this.callback.extraData ?? "0x",
       callerAddress: caller,
-      options: this.params?.options ?? defaultOptions,
-    })
+      options,
+    });
   }
 
   async sendQuery(args: AxiomV2SendQueryArgs): Promise<TransactionReceipt> {
+    if (this.config.privateKey === undefined) {
```

Comment:
We shouldn't do it now, but likely we should have `config` store a `Signer` object instead to support metamask and hardware wallets